### PR TITLE
nimble/ll: Fix aux_data ref counter

### PR DESCRIPTION
**Pull request by rymanluk**
This PR simplifies usage of aux_data ref counter.

### CODE REVIEW

File: nimble/controller/src/ble_ll_scan.c [hunk 1]
```diff
@@ -1294,34 +1278,30 @@ ble_ll_scan_aux_data_ref(struct ble_ll_aux_data *aux_data)
...
 }
 
-int
+void
 ble_ll_scan_aux_data_unref(struct ble_ll_aux_data *aux_data)
 {
     os_sr_t sr;
-    int ret;
 
     if (!aux_data) {
```
Comment (sjanc) [inline nimble/controller/src/ble_ll_scan.c]: I'd make it assert instead of runtime check for NULL

File: nimble/controller/src/ble_ll_scan.c [hunk 2]
```diff
@@ -2903,11 +2868,14 @@ ble_ll_scan_rx_pkt_in(uint8_t ptype, struct os_mbuf *om, struct ble_mbuf_hdr *hd
     struct ble_mbuf_hdr *ble_hdr;
     int ext_adv_mode = -1;
 #if MYNEWT_VAL(BLE_LL_CFG_FEAT_LL_EXT_ADV)
-    struct ble_ll_aux_data *aux_data = hdr->rxinfo.user_data;
+    struct ble_ll_aux_data *aux_data;
     int rc;
     uint8_t evt_possibly_truncated = 0;
 #endif
 
+    aux_data = hdr->rxinfo.user_data;
```
Comment (sjanc) [inline nimble/controller/src/ble_ll_scan.c]: this will not build with EA disabled :)

File: nimble/controller/src/ble_ll_scan.c [hunk 3]
```diff
@@ -3100,7 +3051,8 @@ ble_ll_scan_rx_pkt_in(uint8_t ptype, struct os_mbuf *om, struct ble_mbuf_hdr *hd
             if (BLE_MBUF_HDR_WAIT_AUX(hdr)) {
                 hdr->rxinfo.flags &= ~BLE_MBUF_HDR_F_AUX_PTR_WAIT;
                 if (ble_ll_sched_rmv_elem(&aux_data->sch) == 0) {
-                    ble_ll_scan_aux_data_unref(aux_data);
+                    ble_ll_scan_aux_data_unref(aux_data->sch.cb_arg);
```
Comment (sjanc) [inline nimble/controller/src/ble_ll_scan.c]: this looks odd, what is being unrefed here?

Comment (rymanluk) [reply nimble/controller/src/ble_ll_scan.c]: I've pushed changed code which now have comment about this.  Basically, in here we remove scheduled item which contains its `aux_data` as a `cb_arg` and this is what we want to unref.

Comment (andrzej-kaczmarek) [reply nimble/controller/src/ble_ll_scan.c]: but scheduler in aux_data should point to the same aux_data...

Comment (andrzej-kaczmarek) [reply nimble/controller/src/ble_ll_scan.c]: ok, now I get it (it just explicitly states which ref in urefed)

File: nimble/controller/include/controller/ble_ll_scan.h [hunk 4]
```diff
@@ -88,14 +88,15 @@ struct ble_ll_scan_params
...
-#define BLE_LL_AUX_HAS_ADI              0x80
+#define BLE_LL_AUX_CHAIN_BIT            0x0001
+#define BLE_LL_AUX_INCOMPLETE_BIT       0x0002
+#define BLE_LL_AUX_INCOMPLETE_ERR_BIT   0x0004
+#define BLE_LL_AUX_HAS_ADDRA            0x0008
+#define BLE_LL_AUX_IGNORE_BIT           0x0010
+#define BLE_LL_AUX_HAS_DIR_ADDRA        0x0020
+#define BLE_LL_AUX_TRUNCATED_SENT       0x0040
+#define BLE_LL_AUX_HAS_ADI              0x0080
+#define BLE_LL_SENT_EVENT_TO_HOST       0x0100
```
Comment (sjanc) [inline nimble/controller/include/controller/ble_ll_scan.h]: I'm not sure if I get it why this flag is needed,  some code comment would be nice :)

File: nimble/controller/src/ble_ll_conn.c [hunk 5]
```diff
@@ -2678,22 +2678,25 @@ ble_ll_init_rx_pkt_in(uint8_t pdu_type, uint8_t *rxbuf,
...
-    if (aux_data && !ble_ll_scan_aux_data_unref(aux_data)) {
-        BLE_LL_ASSERT(0);
+     if (ble_hdr->rxinfo.user_data) {
+         /* aux_data just a local helper, no need to ref
+          * as ble_hdr->rxinfo.user_data is unref in the end of this function
+          */
+         aux_data = ble_hdr->rxinfo.user_data;
         return;
-    }
+     }
```
Comment (sjanc) [inline nimble/controller/src/ble_ll_conn.c]: hmm return this doesn't look right

I'd just leave this as
struct ble_ll_aux_data *aux_data = ble_hdr->rxinfo.user_data

File: nimble/controller/src/ble_ll_scan.c [hunk 6]
```diff
@@ -1294,34 +1280,28 @@ ble_ll_scan_aux_data_ref(struct ble_ll_aux_data *aux_data)
...
+void
 ble_ll_scan_aux_data_unref(struct ble_ll_aux_data *aux_data)
 {
     os_sr_t sr;
-    int ret;
 
-    if (!aux_data) {
-        return 0;
-    }
+    BLE_LL_ASSERT(aux_data);
```
[the remark this answers is not in the document]
Comment (rymanluk) [reply nimble/controller/src/ble_ll_scan.c]: looks like rebase issue :) thx

File: nimble/controller/src/ble_ll_scan.c (hunk 2 above)
Comment (rymanluk) [reply nimble/controller/src/ble_ll_scan.c]: are you sure? :)

**Pull request merged**